### ISSUE TITLE
Problem in reproducing Table2 while working on Google Collab

### ISSUE THREAD

**Issue by talrub** [Ping-C/optimizer#3]
Hi,

I uploaded the paper's code to Google Collab in order to restore Table 2 results on  Arch:LeNet Optimizer: G&C .
I run the script : !python train_distributed.py -C configs/table2/mnist_guess.yaml .
I got results for:
num_samples=2 loss_bin= (0.3,0.35) after 320,000 tested models
num_samples=4 loss_bin= (0.3,0.35) after 1,760,000 tested model

But for  num_samples=8 loss_bin= (0.3,0.35), i am not getting result even after  100,000,000 tested models ( The trials continue in an infinite loop).
I was wondering how many tested models did you try until you determined that there are no solutions with 100% training accuracy in a certain loss bin? (i saw in your paper  that you got this case for the linear models).
I did not install your environment.yml in the Google Collab but i don't think that this is what causing the problem.
It will be great if you can advise what should i do in this situation.

Thanks,
Tal

**Comment by Ping-C** [comment]
Hey Tal, thanks for opening this issue. That is quite odd. I know that for some of the cells with samples like 32 at loss loss. It is very difficult to find a solution, but I am surprised that you aren't finding a solution for number_samples=8. Let me look into this.

**Comment by Ping-C** [comment]
Hello Tal, so after some investigation, it seems like the problem is dataseed specific. It seems like for some dataseeds, the G&C model takes a very long time and is never able to find a solution for certain bins. I could reproduce your problem when running exactly a single job.

I previously did not notice this because I usually launch an array of jobs and if some jobs are able to find a model, then I accept the solutions. Now that I think about it, I think my current experiments actually bias the G&C models for certain dataseeds that are easier to find solutions for.... which is not ideal. I have to think a bit more about how this impacts our analysis.

However, if you want to reproduce my previous experimental set up, I would recommend setting a time out for every given dataseed, and if no solution is found after some period, then increment the dataseed. This is not a perfect solution, but is probably the best we can do right now.

**Comment by talrub** [comment]
Hi,
Thanks for the quick response. 
If i want to use the number of tested models as an indication for not being able to find a solution for certain bin, what threshold value do you think will be reasonable?
I was thinking about 3,000,000 tested models per data_seed and trying up to 10 different data seeds until i declare that i am not able to find a solution.
I will point out that after your suggestion regarding the data_seed i got solutions for both num_samples=8 loss_bin= (0.3,0.35) and num_samples=16 loss_bin= (0.3,0.35) but i am still struggling with  num_samples=32 loss_bin= (0.3,0.35).

**Comment by Ping-C** [comment]
For num_sampels=32, it is quite tough. I ran an array of up to 40 jobs at a time, and a majority of the compute is spent on the lowest loss bin with the largest number of samples. What you have sounds reasonable, but I am still unsure as to whether that can give you a solution for the 32 sample/lowest loss bin.

I think the hyperparameter that you have sound reasonable, but in practice, I ran these jobs for maybe up to 24 hours, which may have actually used up to 30m or 100m models.
